lib/chef/knife/rackspace_server_create.rb

```diff
@@ -349,26 +354,29 @@ def run
           load_winrm_deps
         end
 
-        node_name = get_node_name(config[:chef_node_name] || config[:server_name])
-        networks = get_networks(Chef::Config[:knife][:rackspace_networks])
-
-        rackconnect_wait = Chef::Config[:knife][:rackconnect_wait] || config[:rackconnect_wait]
+        node_name                   = get_node_name(config[:chef_node_name] || config[:server_name])
+        networks                    = get_networks(Chef::Config[:knife][:rackspace_networks])
+        rackconnect_wait            = Chef::Config[:knife][:rackconnect_wait] || config[:rackconnect_wait]
+        rackconnect3                = Chef::Config[:knife][:rackconnect_v3_network_id] || config[:rackconnect_v3_network_id]
         rackspace_servicelevel_wait = Chef::Config[:knife][:rackspace_servicelevel_wait] || config[:rackspace_servicelevel_wait]
 
         server = connection.servers.new(
-          :name => node_name,
-          :image_id => Chef::Config[:knife][:image],
-          :flavor_id => locate_config_value(:flavor),
-          :metadata => Chef::Config[:knife][:rackspace_metadata],
-          :disk_config => Chef::Config[:knife][:rackspace_disk_config],
-          :user_data => user_data,
+          :name         => node_name,
+          :image_id     => Chef::Config[:knife][:image],
+          :flavor_id    => locate_config_value(:flavor),
+          :metadata     => Chef::Config[:knife][:rackspace_metadata],
+          :disk_config  => Chef::Config[:knife][:rackspace_disk_config],
+          :user_data    => user_data,
           :config_drive => locate_config_value(:rackspace_config_drive) || false,
-          :personality => files,
-          :key_name => Chef::Config[:knife][:rackspace_ssh_keypair]
+          :personality  => files,
+          :key_name     => Chef::Config[:knife][:rackspace_ssh_keypair]
         )
 
         if version_one?
           server.save
+        elsif rackconnect3
+          nid = Chef::Config[:knife][:rackconnect_v3_network_id]
+          server.save(networks: [nid])
```

Comment:
It seems like adding this logic into the `get_networks` method instead would centralize the conditionals for API version, since both this and the V2 chunk in the `else` below pass the same `networks` argument.